File: src/test/java/easy/FindEvenDigitNumbersTest.java
package easy;

import static org.junit.jupiter.api.Assertions.*;

import java.util.stream.Stream;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

public class FindEvenDigitNumbersTest {
  FindEvenDigitNumbers instance = new FindEvenDigitNumbers();

  @ParameterizedTest
  @MethodSource("dataFindEvenDigitNumbers")
  public void testFindEvenDigitNumbers(int[] nums, int expected) {

    int result = instance.solve(nums);

    assertEquals(expected, result);
  }

  public static Stream<Arguments> dataFindEvenDigitNumbers() {
    return Stream.of(
        Arguments.of(new int[] {12, 345, 2, 6, 7896}, 2),
        Arguments.of(new int[] {555, 901, 482, 1771}, 1));
  }
}

File: easy/find_even_digit_numbers_test.go
package easy

import (
	"testing"
)

func TestFindEvenDigitNumbers(t *testing.T) {
	cases := []struct {
		nums     []int
		expected int
	}{
		{[]int{12, 345, 2, 6, 7896}, 2},
		{[]int{555, 901, 482, 1771}, 1},
	}
	for _, c := range cases {
		t.Run("", func(t *testing.T) {
			result := FindEvenDigitNumbers(c.nums)
			if result != c.expected {
				t.Errorf("FindEvenDigitNumbers(%v) == %d, expected %d", c.nums, result, c.expected)
			}

		})

	}
}
